app/src/main/java/ar/com/wolox/android/example/ui/login/LoginFragment.java

```diff
@@ -16,60 +21,159 @@
 
 import com.google.android.material.textfield.TextInputEditText;
 
+import java.util.regex.Pattern;
+
 /**
-* A simple {@link WolmoFragment} subclass.
-*/
+ * A simple {@link WolmoFragment} subclass.
+ */
 public class LoginFragment extends WolmoFragment implements ILoginView {
-    LoginPresenter presenter;
+    LoginPresenter presenter = new LoginPresenter();
+
+    @BindView(R.id.vLoginButton)
+    Button vLoginButton;
+    @BindView(R.id.vSignUpButton)
+    Button vSignUpButton;
+    @BindView(R.id.vEmailInput)
+    TextInputEditText vEmailInput;
+    @BindView(R.id.vPasswordInput)
+    TextInputEditText vPasswordInput;
+    @BindView(R.id.vTermsConditionsText)
+    TextView vTermsConditionsText;
 
-    @BindView(R.id.vLoginButton) Button vLoginButton;
-    @BindView(R.id.vSignUpButton) Button vSignUpButton;
-    @BindView(R.id.vFirstNameInput) TextInputEditText vFirstNameInput;
-    @BindView(R.id.vLastNameInput) TextInputEditText vLastNameInput;
+    SharedPreferences sharedPref;
+    SharedPreferences.Editor editor;
 
     public int layout() {
         return R.layout.fragment_login;
-    }
+    } // layout()
 
+    /**
+     *
+     */
     public void init() {
         ButterKnife.bind(this, getActivity());
-        vLoginButton.setEnabled(false);
-    }
+        Context context = getActivity();
+
+        sharedPref = context.getSharedPreferences(
```

Comment:
Here you can do getContext().getSharedPref...... It's not really necesary to get context like 55 does